pygmt/datasets/load_earth_dataset.py

```diff
@@ -0,0 +1,210 @@
+    vertical_datum: str
+    horizontal_datum: str
+
+
+datasets = {
+    "earth_relief": Dataset(
+        title="Earth relief",
+        name="elevation",
+        long_name="elevation relative to the geoid",
+        units="meters",
+        vertical_datum="EMG96",
+        horizontal_datum="WGS84",
+        resolutions={
+            "01d": Resolution(["pixel", "gridline"], False),
+            "30m": Resolution(["pixel", "gridline"], False),
+            "20m": Resolution(["pixel", "gridline"], False),
+            "15m": Resolution(["pixel", "gridline"], False),
+            "10m": Resolution(["pixel", "gridline"], False),
+            "06m": Resolution(["pixel", "gridline"], False),
+            "05m": Resolution(["pixel", "gridline"], True),
+            "04m": Resolution(["pixel", "gridline"], True),
+            "03m": Resolution(["pixel", "gridline"], True),
+            "02m": Resolution(["pixel", "gridline"], True),
+            "01m": Resolution(["pixel", "gridline"], True),
+            "30s": Resolution(["pixel", "gridline"], True),
+            "15s": Resolution(["pixel"], True),
+            "03s": Resolution(["gridline"], True),
+            "01s": Resolution(["gridline"], True),
+        },
+    ),
+    "earth_age": Dataset(
+        title="seafloor age",
+        name="seafloor_age",
+        long_name="age of seafloor crust",
+        units="Myr",
+        vertical_datum="EMG96",
+        horizontal_datum="WGS84",
+        resolutions={
+            "01d": Resolution(["pixel", "gridline"], False),
+            "30m": Resolution(["pixel", "gridline"], False),
+            "20m": Resolution(["pixel", "gridline"], False),
+            "15m": Resolution(["pixel", "gridline"], False),
+            "10m": Resolution(["pixel", "gridline"], False),
+            "06m": Resolution(["pixel", "gridline"], False),
+            "05m": Resolution(["pixel", "gridline"], True),
+            "04m": Resolution(["pixel", "gridline"], True),
+            "03m": Resolution(["pixel", "gridline"], True),
+            "02m": Resolution(["pixel", "gridline"], True),
+            "01m": Resolution(["gridline"], True),
+        },
+    ),
+}
+
+
+@kwargs_to_strings(region="sequence")
+def _load_earth_dataset(dataset_name, dataset_prefix, resolution, region, registration):
+    r"""
+    Load Earth datasets.
+
+    Parameters
+    ----------
+    dataset_name : str
+        The name for the dataset in the 'datasets' dictionary.
+
+    dataset_prefix : str
+        The prefix for the dataset that will be passed to the GMT C API.
+
+    resolution : str
+        The grid resolution. The suffix ``d`` and ``m`` stand for
+        arc-degree and arc-minute.
+
+    region : str or list
+        The subregion of the grid to load, in the forms of a list
+        [*xmin*, *xmax*, *ymin*, *ymax*] or a string *xmin/xmax/ymin/ymax*.
+        Required for grids with resolutions higher than 5
+        arc-minute (i.e., ``"05m"``).
```

Comment:
For some datasets, high resolutions grids are not tiled.
```suggestion
        Required for tiled grids. 
```